qml/MainMenu.qml

```diff
@@ -90,16 +90,15 @@ Item {
                 CheckBox {
                     id: autoGainChB
                     text: "Auto Gain"
-                    checked: ping.modeAuto
+                    checked: ping.mode_auto
                     onCheckedChanged: {
-                        //setEchosounderAuto(checked)
-                        ping.request(1102)
+                        ping.mode_auto = checked
```

Comment:
You need to request the mode, otherwise will not be updated in the interface if something goes wrong.